isso/ext/notifications.py

```diff
@@ -139,17 +139,43 @@ def format(self, thread, comment, parent_comment, recipient=None, admin=False):
 
         else:
             uri = self.public_endpoint + "/id/%i" % parent_comment["id"]
-            key = self.isso.sign(('unsubscribe', recipient))
+            key = self.isso.sign(('unsubscribe', recipient["email"]))
 
-            rv.write("Unsubscribe from this conversation: %s\n" % (uri + "/unsubscribe/" + quote(recipient) + "/" + key))
+            rv.write("Unsubscribe from this conversation: %s\n" % (uri + "/unsubscribe/" + quote(recipient["email"]) + "/" + key))
 
         rv.seek(0)
         return rv.read()
 
+    def notify_subject(self, thread, comment, parent_comment=None, recipient=None, admin=False):
+        if admin:
+            return self.isso.conf.get("mail", "subject_admin").format(
+                title=thread["title"],
+                replier=comment["author"] or "Anonymous")
+        else:
+            subject_format = list(self.isso.conf.getiter("mail", "subject_user"))
+            if len(subject_format) == 1:
+                return subject_format[0].format(
+                    title=thread["title"],
+                    repliee=parent_comment["author"] or "Anonymous",
+                    replier=comment["author"] or "Anonymous",
+                    receiver=recipient["author"] or "Anonymous")
```

Comment:
How can receiver be anonymous?

Reply:
That's in a title. A receiver can be named "Anonymous" without a name.